### PR TITLE
chore(presto): Expose schema and indexes to _partition_query method

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR exposes the schema and indexes to the internal (private) `PrestoBaseEngineSpec._partition_query` method as these could be useful for custom overrides of the partition query.

### TESTING INSTRUCTIONS

CI.
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/presto.py [hunk 1]
```diff
@@ -423,24 +423,27 @@ def get_function_names(cls, database: Database) -> List[str]:
         return database.get_df("SHOW FUNCTIONS")["Function"].tolist()
 
     @classmethod
-    def _partition_query(  # pylint: disable=too-many-arguments,too-many-locals
+    def _partition_query(  # pylint: disable=too-many-arguments,too-many-locals,unused-argument
         cls,
+        schema: Optional[str],
```
Comment (john-bodley) [inline superset/db_engine_specs/presto.py:428]: Given this is a private method, changing the order of the function arguments is deemed non-breaking.

File: superset/db_engine_specs/presto.py [hunk 2]
```diff
@@ -423,24 +423,30 @@ def get_function_names(cls, database: Database) -> List[str]:
...
         database: Database,
         limit: int = 0,
         order_by: Optional[List[Tuple[str, bool]]] = None,
         filters: Optional[Dict[Any, Any]] = None,
     ) -> str:
-        """Returns a partition query
+        """
+        Return a partition query.
 
+        Note the unused arguments are exposed for sub-classing purposes where custom
```
Comment (john-bodley) [inline superset/db_engine_specs/presto.py:439]: @michael-s-molina per our offline discussion I added this comment as to why the unused arguments are included.

File: superset/db_engine_specs/presto.py [hunk 3]
```diff
@@ -423,24 +423,30 @@ def get_function_names(cls, database: Database) -> List[str]:
...
         limit: int = 0,
         order_by: Optional[List[Tuple[str, bool]]] = None,
         filters: Optional[Dict[Any, Any]] = None,
     ) -> str:
-        """Returns a partition query
+        """
+        Return a partition query.
+
+        Note the unused arguments are exposed for sub-classing purposes where custom
+        integrations may require the schema, indexes, etc. to build the paritiion query.
```
Comment (michael-s-molina) [inline superset/db_engine_specs/presto.py:440]:
```suggestion
        integrations may require the schema, indexes, etc. to build the partition query.
```

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**